client/src/utils/CardContext.js

```diff
@@ -0,0 +1,32 @@
+import React, { useState, createContext, useContext, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { UsernameContext } from './UsernameContext';
+
+export const CardContext = createContext();
+
+export const CardProvider = ({ children }) => {
```

Comment:
I still think the naming should be focused on coins/coin data not cards, since you generate cards from the data at a lower level and you provide the data necessary to do so at this level